external-testing/README.md

```diff
@@ -0,0 +1,65 @@
+# External Testing
+
+Let's talk about testing process against external resources. What does that
+mean? Well, talking to GitHub API, COPR API, running `fedpkg`, cloning
+repositories...
+
+
+## Use cases
+
+* Interact with an external API (such as GitHub, COPR, Pagure, FAS, k8s)
+  * Authenticated vs. unauthenticated calls.
+* Clone a git repository.
+* Construct a git repository.
+* Run a command to perform an action.
+  * Changes happening only locally.
+  * RO access to a remote service.
+  * R/W access to a remote service.
+* Combination of multiple use cases into a single test case.
+
+
+## Solutions
+
+* flexmock
+
+* pytest fixtures
+
+* requre
+
+* VCRpy
+
+* Betamax
+
+
+## VCRpy vs. requre
+
+| A feature | VCRpy | requre |
+|-----------|-------|--------|
+| Record & replay for python requests | ✓ | ✓ |
+| Match on call stack | ✘ | ✓ |
+| Match on URL & request type | ✓ | ✘ |
+| Calling the same URL twice works as expected | ✓ | ✘ |
```

Comment:
it is also possible. it was the first implementation what were in `OGR` it used `URL` and `data` keys for storing responses.
It is also possible now, but it is not the default mode. it were changed to use list by default, instead of just dict, what matched results.

Reply:
Yes, @jscotka is right. It was common in OGR to save a list of responses (exactly the situation I was asking Tomáš) so we made it the default.